### ISSUE TITLE
Can you give your approval for this description of DeepLIFT as implemented in the DeepExplain repo?

### ISSUE THREAD

**Issue by AvantiShri** [marcoancona/DeepExplain#14]
Hello,

Thank you for this wonderful implementation. I'm the author of DeepLIFT, and I wanted to augment DeepLIFT's [FAQ](https://github.com/kundajelab/deeplift/blob/master/README.md#how-does-the-implementation-in-this-repository-compare-with-ancona-et-al-iclr-2018) section to compare the implementation there with the implementation in this repo. This is what I have - can you give your approval or suggest edits?

Ancona et al., authors of the [DeepExplain](https://github.com/marcoancona/DeepExplain) repository, leveraged overriding of the gradient operators in Tensorflow to implement the Rescale rule of DeepLIFT. Their implementation can work with a wider variety of architectures than the DeepLIFT implementation in this repository, and is potentially more computationally efficient, but it does not have the advantages of the RevealCancel rule (which deals with failure modes such as the min function). Note that their implementation can work with architectures that DeepLIFT was not designed for, such as LSTMs and GRUs (their implementation would use the standard gradient backpropagation rule in all cases where the gradient operator was not overridden). We have not studied the appropriateness of this approach, but the authors did find that “Integrated Gradients and DeepLIFT have very high correlation, suggesting that the latter is a good (and faster) approximation of the former in practice”.

**Comment by marcoancona** [comment]
Hi Avanti!
Thanks for reaching out. Sure you can put this in your FAQ section. I would also mention that if it is used with multiplicative units (ie. gates) it might lead to unexpected results and the property of summation-to-delta does not hold anymore. Adapting to LSTMs or other architectures that use gates would require to make some different choices in the backprogation procedure instead of blindly apply the chain rule. For example as Arras et al. did for LRP in https://arxiv.org/abs/1612.07843

**Comment by AvantiShri** [comment]
Thanks - yes I will mention that.
